File: zig/resistor-color/resistor_color.zig
pub const ColorBand = enum(u4) {  black, 
                        brown, 
                        red, 
                        orange, 
                        yellow,
                        green, 
                        blue, 
                        violet, 
                        grey, 
                        white, 
                    };

pub fn colorCode(color: ColorBand) isize {
    return @enumToInt(color);
}

pub fn colors() []const ColorBand {
    return &[_]ColorBand {
        .black, .brown, .red, .orange, .yellow,
        .green, .blue, .violet, .grey, .white,
    };
}

File: julia/resistor-color/resistor-color.jl
colors_vec = [
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white",
        ]

colorcode(color) = findfirst(x -> x == color, colors_vec) - 1

colors() = colors_vec
